File: engine/src/main/java/com/terralite/engine/input/InputAction.java
package com.terralite.engine.input;

import java.util.Objects;

public record InputAction(String name) {
    public InputAction {
        if (Objects.requireNonNull(name, "name").isBlank()) {
            throw new IllegalArgumentException("Input action name cannot be blank");
        }
    }

    public static InputAction of(String name) {
        return new InputAction(name);
    }
}

File: engine/src/commonMain/kotlin/com/terralite/engine/input/InputAction.kt
package com.terralite.engine.input

@JvmRecord
data class InputAction(val name: String) {
    init {
        require(name.isNotBlank()) { "Input action name cannot be blank" }
    }

    companion object {
        @JvmStatic fun of(name: String): InputAction = InputAction(name)
    }
}
